### PR TITLE
Rename "distribution" to "operating system type"

### PR DESCRIPTION
**Pull request by pellared**
## Why

"Distribution" means something else in OTel sense: https://opentelemetry.io/docs/concepts/distributions/

Reference https://github.com/open-telemetry/opentelemetry.io/issues/1689#issuecomment-1262808129

## What

Rename "distribution" to "operating system type"

## Tests

CI

## Checklist

- [x] ~~`CHANGELOG.md` is updated.~~
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -99,16 +99,16 @@ environment variables as parameters:
 
 | Parameter      | Description                                                      | Required | Default value                                                                     |
 |----------------|------------------------------------------------------------------|----------|-----------------------------------------------------------------------------------|
-| `DISTRIBUTION` | Possible values: `linux-glibc`, `linux-musl`, `macos`, `windows` | Yes      |                                                                                   |
 | `INSTALL_DIR`  | Location where binaries are to be installed                      | No       | `./otel-dotnet-auto`                                                              |
+| `OS_TYPE`      | Possible values: `linux-glibc`, `linux-musl`, `macos`, `windows` | Yes      |                                                                                   |
```
Comment (theletterf) [inline docs/README.md:103]: Why no default? Possibly a silly question. Alternatively, I'd add a sentence in that column stating how the value gets filled.

File: docs/README.md [hunk 2]
```diff
@@ -179,15 +179,15 @@ environment variables as parameters:
 
 | Parameter          | Description                                                            | Required | Default value        |
 |--------------------|------------------------------------------------------------------------|----------|----------------------|
-| `DISTRIBUTION`     | Possible values: `linux-glibc`, `linux-musl`, `macos`, `windows`       | Yes      |                      |
 | `ENABLE_PROFILING` | Whether to set the .NET CLR Profiler, possible values: `true`, `false` | No       | `true`               |
 | `INSTALL_DIR`      | Location where binaries are to be installed                            | No       | `./otel-dotnet-auto` |
+| `OS_TYPE`          | Possible values: `linux-glibc`, `linux-musl`, `macos`, `windows`       | Yes      |                      |
```
Comment (theletterf) [inline docs/README.md:184]: Same as above.

File: instrument.sh [hunk 3]
```diff
@@ -1,11 +1,11 @@
 #!/bin/sh
 
 # validate input
-case "$DISTRIBUTION" in
+case "$OS_TYPE" in
   "linux-glibc"|"linux-musl"|"macos"|"windows")
     ;;
   *)
-    echo "Please specify the distribution by setting the DISTRIBUTION env var. Supported values: linux-glibc, linux-musl, macos, windows." >&2
+    echo "Please specify the operating system type by setting the OS_TYPE environment variable. Supported values: linux-glibc, linux-musl, macos, windows." >&2
```
Comment (theletterf) [inline instrument.sh:8]:
```suggestion
    echo "Set the operating system type using the OS_TYPE environment variable. Supported values: linux-glibc, linux-musl, macos, windows." >&2
```

**Review by theletterf — APPROVED** [review]

File: docs/README.md (hunk 1 above)
Comment (pellared) [reply docs/README.md:103]: Good question.

I thought about setting `linux-glibc` as default. However, when one will execute it on another OS (e.g. Windows), she/he will have a really hard time understanding why the instrumentation is not working.

Calculating the default value based on the machine info is hard and I do not want us to maintain this code.

**Pull request merged**